bin/zerobufCxx.py

```diff
@@ -832,23 +915,63 @@ def parse_members(self, fbsFile):
             is_byte_type = fbs_type == "byte" or fbs_type == "ubyte"
             value_type = ValueType(cxxtype, cxxtype_size, is_zerobuf_type, is_enum_type, is_byte_type)
 
+            properties[name] = {}
+            property = properties[name]
+
             if self.is_dynamic(attrib, fbsFile):
                 if len(attrib) == 2 and is_zerobuf_type:
                     member = DynamicZeroBufMember(name, value_type, dynamic_type_index)
+                    table = next(x for x in fbsFile.tables if x.name == cxxtype)
+                    properties[name] = table.json_schema
                 else:
                     member = DynamicMember(name, value_type, dynamic_type_index, self.name)
+                    if is_byte_type: # byte arrays are base64 strings
+                        property['type'] = 'string'
+                        property['media'] = {}
+                        property['media']['binaryEncoding'] = 'base64'
+                    elif fbs_type == 'string':
+                        property['type'] = 'string'
+                    else:
+                        property['type'] = 'array'
+                        if is_zerobuf_type:
+                            # array of ZeroBuf objects
+                            table = next(x for x in fbsFile.tables if x.name == cxxtype)
+                            property['items'] = table.json_schema
+                        else:
+                            # POD array
+                            property['items'] = {}
+                            property['items']['type'] = fbs_to_json_type(fbs_type)
+
                 dynamic_type_index += 1
                 self.dynamic_members.append(member)
             else:
                 if len(attrib) == 2 or len(attrib) == 3:
                     member = FixedSizeMember(name, value_type)
+                    if is_zerobuf_type:
+                        table = next(x for x in fbsFile.tables if x.name == cxxtype)
+                        properties[name] = table.json_schema
+                    elif is_enum_type:
+                        enum = next(x for x in fbsFile.enums if x.name == cxxtype)
+                        properties[name] = enum.json_schema
+                    else:
+                        property['type'] = fbs_to_json_type(fbs_type)
                     if len(attrib) == 3:
                         default_values = attrib[2]
                         setter = "set{0}({1}( {2} ));".format(member.cxxName, cxxtype, default_values)
                         self.default_value_setters.append(setter)
                 else:
                     elem_count = int(attrib[4])
                     member = FixedSizeArray(name, value_type, elem_count, self.name)
+                    if is_byte_type:
+                        property['type'] = 'string'
+                        property['media'] = {}
+                        property['media']['binaryEncoding'] = 'base64'
```

Comment:
these three lines are a copy-paste of the other "if is_byte_type" above, maybe can be combined somehow?